### PR TITLE
[ISSUE-298] Enable volume expansion tests

### LINKED ISSUE #298: Volume expansion test coverage
Make sanity and e2e tests for volume expansion work

### PR DESCRIPTION
**Pull request by rkiselenko**
### Resolves #298

Enable volume expansion tests

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

### CODE REVIEW

File: test/e2e/scenarios/bare-metal-testdriver.go [hunk 1]
```diff
@@ -120,6 +126,11 @@ func PrepareCSI(d *baremetalDriver, f *framework.Framework, deployConfig bool) (
 
 	return testConf, func() {
 		// wait until ephemeral volume will be deleted
+		if testConf.Framework.BaseName == volumeExpandTag {
+			cm := d.constructDefaultLoopbackConfig(testConf.Framework.Namespace.Name)
+			err := testConf.Framework.ClientSet.CoreV1().ConfigMaps(testConf.Framework.Namespace.Name).Delete(context.TODO(), cm.Name, metav1.DeleteOptions{})
```
Comment (safronovD) [inline test/e2e/scenarios/bare-metal-testdriver.go:131]: Why `delete`? :)

File: test/e2e/scenarios/bare-metal-testdriver.go [hunk 2]
```diff
@@ -91,11 +100,15 @@ func (d *baremetalDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern)
 
 		// We have volume and exec pvc test for default fs (equals to xfs) in short CI
 		// Not need to perform them for other filesystems
-		if pattern.FsType == "xfs" || pattern.FsType == "ext4" || pattern.FsType == "ext3" {
+		if pattern.FsType == xfsFs || pattern.FsType == ext4Fs || pattern.FsType == ext3Fs {
 			e2eskipper.Skipf("Should skip tests in short CI suite -- skipping")
 		}
 	}
 
+	if pattern.AllowExpansion && pattern.VolMode == corev1.PersistentVolumeBlock {
```
Comment (safronovD) [inline test/e2e/scenarios/bare-metal-testdriver.go:108]: Is it works for short-ci suite without skipping expansion tests?

File: pkg/node/Dockerfile-kernel-5.4.build [hunk 3]
```diff
@@ -3,7 +3,4 @@ FROM    ubuntu:21.04
 ADD     health_probe    health_probe
 
 # On Ubuntu 21.04 fdisk is not installed by defaul
-# Remove bash packet to get rid of related CVEs
-RUN     apt update --no-install-recommends -y -q; apt remove --no-install-recommends -y --allow-remove-essential -q bash; apt install --no-install-recommends -y -q util-linux parted xfsprogs lvm2 fdisk gdisk strace udev net-tools
-
-
+RUN     apt update --no-install-recommends -y -q; apt install --no-install-recommends -y -q util-linux parted xfsprogs lvm2 fdisk gdisk strace udev net-tools
```
Comment (safronovD) [inline pkg/node/Dockerfile-kernel-5.4.build:6]: Should we place comment here about unstable bash version?

File: test/kind/kind.yaml [hunk 4]
```diff
@@ -1,67 +1,67 @@
...
-      - hostPath: /dev
-        containerPath: /dev
-      - hostPath: /run/udev
-        containerPath: /run/udev
-      - hostPath: /run/lvm
-        containerPath: /run/lvm
-      - hostPath: /run/lock
-        containerPath: /run/lock
\ No newline at end of file
+# the control plane node config
```
Comment (safronovD) [inline test/kind/kind.yaml:5]: Why do you remove indents?

File: test/e2e/scenarios/bare-metal-testdriver.go [hunk 5]
```diff
@@ -193,10 +224,12 @@ func (d *baremetalDriver) constructDefaultLoopbackConfig(namespace string) *core
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      cmName,
 			Namespace: namespace,
+			Labels:    map[string]string{"app": "csi-baremetal-node"},
 		},
 		Data: map[string]string{
 			"config.yaml": "\n" +
-				"defaultDrivePerNodeCount: 3\n" +
+				fmt.Sprintf("defaultDriveSize: %s\n", maxDriveSize) +
```
Comment (safronovD) [inline test/e2e/scenarios/bare-metal-testdriver.go:231]: Isn't it better to stay '100Mi` as default?

File: pkg/node/Dockerfile-kernel-5.4.build (hunk 3 above)
Comment (rkiselenko) [reply pkg/node/Dockerfile-kernel-5.4.build:6]: What version is unstable? There is no hardcoded version.

File: test/kind/kind.yaml (hunk 4 above)
Comment (rkiselenko) [reply test/kind/kind.yaml:5]: That the difference?

File: test/e2e/scenarios/bare-metal-testdriver.go (hunk 2 above)
Comment (rkiselenko) [reply test/e2e/scenarios/bare-metal-testdriver.go:108]: AFAIK short-ci isn't use community tests

File: pkg/node/Dockerfile-kernel-5.4.build (hunk 3 above)
Comment (safronovD) [reply pkg/node/Dockerfile-kernel-5.4.build:6]: I'm not sure, why did we delete it?

**Review by safronovD — APPROVED** [review]

File: pkg/node/Dockerfile-kernel-5.4.build (hunk 3 above)
Comment (rkiselenko) [reply pkg/node/Dockerfile-kernel-5.4.build:6]: Because  of `bash`, `bash` required in order to work with lvg

Comment (mishoyama) [reply pkg/node/Dockerfile-kernel-5.4.build:6]: @rkiselenko there is a CVE associated with the `bash` package - https://ubuntu.com/security/CVE-2019-18276. We need to make a note here why we need to have `bash` and create a separate task to address CVE.

File: pkg/node/Dockerfile.build [hunk 6]
```diff
@@ -2,7 +2,4 @@ FROM    ubuntu:18.04
 
 ADD     health_probe    health_probe
 
-# Remove bash packet to get rid of related CVEs
```
Comment (mishoyama) [inline pkg/node/Dockerfile.build:5]: Same here

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: pkg/node/Dockerfile.build (hunk 6 above)
Comment (rkiselenko) [reply pkg/node/Dockerfile.build:5]: 21.04 Not vulnerable accouding to https://ubuntu.com/security/CVE-2019-18276 
image

**Review by mishoyama — APPROVED** [review]

**Review by mitsaa — APPROVED** [review]

**Review by Aber4Nod — APPROVED** [review]

**Pull request merged**